packages/loaders/src/elements/Dots.tsx

```diff
@@ -39,14 +40,23 @@ export interface IDotsProps extends React.HTMLAttributes<SVGSVGElement> {
 }
 
 const Dots: React.FC<IDotsProps> = ({ size, color, duration, delayMS, ...other }) => {
-  const noAnimatedSVGSupport = useCSSSVGAnimation();
-  const { delayComplete } = useSchedule({ duration, delayMS, loop: noAnimatedSVGSupport });
+  const theme = useContext(ThemeContext);
+  const environment = useDocument(theme);
+  const canTransformSVG = useRef<boolean | null>(null);
```

Comment:
For this to update the effect below, should this be `useState`? It could then be included as a dependency for `useEffect`.

Reply:
There should not be a need to sync this value with the effect below because the effect runs on each render.

Reply:
Oh cool. It may be useful to still place the original calculation in an effect then. 

The component doesn't need to calculate `canTransformSVG` on every render. An environment shouldn't change this ability after the first calculation.

Reply:
Does the `null` check on line 47 prevent unnecessary calculations?

```jsx
if (environment && canTransformSVG.current === null) {
  const svg = environment.createElementNS('http://www.w3.org/2000/svg', 'svg');

  // Once ref is set to true or false, additional renders will not hit this branch again.
  canTransformSVG.current = 'transform' in svg;
}
```

Reply:
LGTM!